src/components/Image/index.tsx

```diff
@@ -33,17 +31,14 @@ function Image(props) {
     useEffect(() => {
         // If an onLoad callback was specified then manually call it and pass
         // the natural image dimensions to match the native API
-        if (onLoad == null) {
+        if (typeof onLoad !== 'function' || typeof source === 'number' || !source.uri) {
```

Comment:
Same, why was this condition changed?

Reply:
So, regarding `onLoad`, you can take a look at [Bartosz's comment](https://github.com/Expensify/App/pull/31296#discussion_r1392877134), the other additions to this condition are to make sure `source.uri` can be accessed on **line 37**. Hope that answers your question!